staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -222,8 +225,13 @@ func (r *crdHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	crdTenant := requestInfo.Tenant
```

Comment:
Should this be controlled by an API Server parameter option? It should be treated as an error if the option is not turned on, right? It's kind of allowing anonymous tenant or not.

Reply:
Also, should this be included in the NormalizeTenant() method?

Reply:
The tenant in RequestInfo is empty if:
1. a backward-compatible crd yaml is used, where the tenant in ObjectMeta is empty
2. the user is system user.

NormalizeTenant()  func can not set the empty tenant in the request info in a short-path request if the user is a system user, as the system users are allowed to perform operations across all tenants,  and the empty tenant here may mean TenantAll.

For non-CRD requests, the API server will set the tenant after it figures out it is a request on a specific object, instead of a request across all the tenants. It is done in func ValidTenant() in meta.go. The code here is the counterpart in apiextension server. 

I added more comments in the code to clarify.